squad/core/models.py

```diff
@@ -232,12 +232,22 @@ def test_suites_by_environment(self):
         result = OrderedDict()
         envlist = set([t.environment for t in test_runs])
         for env in sorted(envlist, key=lambda env: env.slug):
-            result[env] = set()
+            result[env] = dict()
         for tr in test_runs:
             for t in tr.tests.all():
-                result[tr.environment].add(t.suite)
+                if t.suite in result[tr.environment].keys():
+                    if t.status in result[tr.environment][t.suite].keys():
+                        result[tr.environment][t.suite][t.status] += 1
+                    else:
+                        result[tr.environment][t.suite].update({t.status: 1})
+                else:
+                    result[tr.environment].update({t.suite: {t.status: 1}})
```

Comment:
@you might want to use default values to simplify this. e.g. I think this entire block could be replaced with the following (untested):

```python
result[tr.environment].setdefault(t.suite, { True: 0, False: 0, None: 0 })
result[tr.environment][t.suite][t.status] += 1
```

Reply:
I expect we would get comment that zeros should not be listed in the notification. Probably could be sorted in the email template though.

Reply:
Fixed